### DISCUSSION TITLE
Why both os.Interrupt and syscall.SIGTERM

### DISCUSSION THREAD

**Discussion by gcstang** [MarioCarrion/todo-api-microservice-example#20] [Q&A] [answered]
os.Interrupt is a variable to syscall.SIGTERM or incase of Plan9 syscall.Note("interrupt")

In reference to:
`ctx, stop := signal.NotifyContext(context.Background(),
		os.Interrupt,
		syscall.SIGTERM,
		syscall.SIGQUIT)`

**Comment by MarioCarrion** [comment, marked as answer]
Hello @gcstang 

* `os.Interrupt` is an "alias" to `syscall.SIGINT`: https://pkg.go.dev/os#Interrupt ; it's used to support stopping the process locally, typically when using `CONTROL+C`.
* `syscall.SIGTERM` is the default signal sent by docker to the main process to stop it https://docs.docker.com/engine/reference/commandline/stop/ ; other services like ([AWS ECS](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_StopTask.html)) also use the same signal to stop running containers.

The idea with supporting both is to enable the local development as well as the future use case where the process is running in a docker container.

Were you suggesting to explicitly indicate `syscall.SIGINT` instead, to make things clearer? If so I think that change is reasonable considering this is only building binaries for Linux at the moment.

**Comment by gcstang** [comment]
in the example you have all 3 instead of just two,I was wondering why
instead of just SIGINT and SIGTERM
or os.Interrupt and SIGTERM
